File: programmers/Level1/CPP/소수 찾기.cpp
using namespace std;

int solution(int n) {
    int answer = n-1;
    bool arr[1000001]={};
    
    for(int i = 2; i<=n; i++){
        if(arr[i] == 1) continue;
        for(int j = 2; j<=n; j++){
            if(j*i > n) break;
            if(arr[j*i] == 0){
                arr[j*i]=1;
                answer--;
            }
        }
    }
    
    return answer;
}

File: programmers/Level1/JAVA/소수 찾기.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        
        boolean[] arr = new boolean[1000000];
        
        for(int i = 2; i <= n; i++){
            int j = 2;
            while(j * i <= n){
                arr[j*i] = true;
                j += 1;
            }
        }
        
        for(int i = 2;  i<=n;i++){
            if(arr[i] == false) answer++;
        }
        
        return answer;
    }
}
